### ISSUE TITLE
Worse performance after converting tf.compat.v1.nn.ctc_loss to tf.nn.ctc_loss

### ISSUE THREAD

**Issue by thetruejacob** [tensorflow/tensorflow#53076]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 20.04
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): v2.5.0-rc3-213-ga4dfb8d1a71
- Python version: 3.8.10
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 1..2
- GPU model and memory: GeForce RTX 3070 8GB

**Describe the current behavior**
I am trying to convert parts of my code from using tf.compat.v1 to pure tf functions. Here is the previous working function, that gives me good performance.

`self.loss = tf.reduce_mean(
            input_tensor=tf.compat.v1.nn.ctc_loss(
                labels=self.gt_texts,
                inputs=self.ctc_in_3d_tbc,
                sequence_length=self.seq_len,
                ctc_merge_repeated=True,
            )
        )
`

This is my attempt to rewrite the function using tf.nn.ctc_loss.

`self.loss = tf.reduce_mean(
            input_tensor=tf.nn.ctc_loss(
                labels=self.gt_texts,  # sparse tensor
                logits=self.ctc_in_3d_tbc,  
                label_length=None,  
                logit_length=self.seq_len,  
                blank_index=-1,
            ))
`
Please let me know if attaching logfiles will be helpful. I figure I must simply be configuring some of the arguments wrong.

**Comment by sushreebarsa** [comment]
@thetruejacob Could you please try to execute your code on latest stable version of TF `2.7.0` and let us know if the issue still persists? Thank you!

**Comment by thetruejacob** [comment]
@sushreebarsa Thank you for the quick reply. I'm currently running on TF2.7.0 and can say that while I used to get around 60-63% accuracies, now I only get 46-48% with the new ctc loss function. Am I passing in the right function arguments?

**Comment by sushreebarsa** [comment]
@thetruejacob 
In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!
